Index: api/src/main/java/com/globalog/api/domain/Wallet.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/api/src/main/java/com/globalog/api/domain/Wallet.java b/api/src/main/java/com/globalog/api/domain/Wallet.java
new file mode 100644
--- /dev/null	(date 1771951052692)
+++ b/api/src/main/java/com/globalog/api/domain/Wallet.java	(date 1771951052692)
@@ -0,0 +1,34 @@
+package com.globalog.api.domain;
+
+import jakarta.persistence.*;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import java.math.BigDecimal;
+
+@Entity
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class Wallet {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false, length = 3)
+    private String currency;
+
+    @Column(nullable = false, precision = 15, scale = 2)
+    private BigDecimal balance;
+
+    @Column(nullable = false, precision = 15, scale = 2)
+    private BigDecimal averageExchangeRate;
+
+    @Builder
+    public Wallet(String currency, BigDecimal balance, BigDecimal averageExchangeRate) {
+        this.currency = currency;
+        this.balance = balance;
+        this.averageExchangeRate = averageExchangeRate;
+    }
+}
